modules/foreman_host.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2017, Ismael Puerto <ipuertofreire@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_host
+short_description: Manage Foreman Host
+description:
+    - Manage host
+author: "Ismael Puerto (@ismaelpuerto)"
+requirements:
+    - "nailgun >= 0.28.0"
+    - "python >= 2.6"
+options:
+    server_url:
+        description:
+            - URL of Foreman server
+        required: true
+    username:
+        description:
+            - Username on Foreman server
+        required: true
+    password:
+        description:
+            - Password for user accessing Foreman server
+        required: true
+    verify_ssl:
+        description:
+            - Verify SSL of the Foreman server
+        required: false
+    hostname:
+        description:
+            - Hostname
+        required: true
+    mac:
+        description:
+            - MAC HOST
+        required: true
```

Comment:
For VM eg mac is not required.
But as far as I understand you are so far only supporting bare-metal deployments?
Maybe we should try and immediately include at least also VM, since that is probably the most used provider.